qulice-findbugs/src/test/java/com/qulice/findbugs/FindBugsExcludesTest.java

```diff
@@ -0,0 +1,262 @@
+        xml.assertXpath("count(/FindBugsFilter/Match[Class[@name='Foo']]/Bug)=0");
+    }
+
+    @Test
+    public void multipleClasses() throws IOException {
+        final ExcludesXml xml = new ExcludesXml("Alice:set:FI_EMPTY,Bob:get");
+        xml.assertXpath("count(//Match)=2");
+        xml.assertXpath("count(//Class)=2");
+        xml.assertXpath("count(//Method)=2");
+        xml.assertXpath("count(//Bug)=1");
+        xml.assertXpath("count(//Match/Class[@name='Alice'])=1");
+        xml.assertXpath("count(//Match[Class[@name='Alice']]/Method[@name='set'])=1");
+        xml.assertXpath("count(//Match[Class[@name='Alice'] and Method[@name='set']]/Bug[@pattern='FI_EMPTY'])=1");
+        xml.assertXpath("count(//Match/Class[@name='Bob'])=1");
+        xml.assertXpath("count(//Match[Class[@name='Bob']]/Method[@name='get'])=1");
+        xml.assertXpath("count(//Match[Class[@name='Bob']]/Bug)=0");
+    }
+
+    @Test
+    public void classMethod() throws IOException {
+        final ExcludesXml xml = new ExcludesXml("MyClass:myMethod");
+        xml.assertXpath("count(/FindBugsFilter/Match[Class[@name='MyClass']]/Method[@name='myMethod'])=1");
+    }
+
+    @Test
+    public void classMethodRule() throws IOException {
+        final ExcludesXml xml = new ExcludesXml("MyClass:myMethod:DLS_DEAD_LOCAL_STORE");
+        xml.assertXpath("count(//Match)=1");
+        xml.assertXpath("count(//Match/Class[@name='MyClass'])=1");
+        xml.assertXpath("count(//Match[Class[@name='MyClass'] and Method[@name='myMethod']]/Bug[@pattern='DLS_DEAD_LOCAL_STORE'])=1");
+    }
+
+    @Test
+    public void wholeClassOneRule() throws IOException {
+        final ExcludesXml xml = new ExcludesXml("MyClass::FI_USELESS");
+        xml.assertXpath("count(//Match)=1");
+        xml.assertXpath("count(//Match/Class[@name='MyClass'])=1");
+        xml.assertXpath("count(//Match/Method)=0");
+        xml.assertXpath("count(//Match/Bug[@pattern='FI_USELESS'])=1");
+    }
+
+    @Test
+    public void anyClassOneRule() throws IOException {
+        final ExcludesXml xml = new ExcludesXml("::DM_EXIT");
+        xml.assertXpath("count(//Class)=0");
+        xml.assertXpath("count(//Method)=0");
+        xml.assertXpath("count(//Match)=1");
+        xml.assertXpath("count(//Match/*)=1");
+        xml.assertXpath("count(//Match/Bug[@pattern='DM_EXIT'])=1");
+    }
+
+    @Test
+    public void tooManyColonsAreIgnored() throws IOException {
+        final ExcludesXml xml = new ExcludesXml("MyClass:myMethod:DM_EXIT:::");
+        xml.assertXpath("count(/FindBugsFilter)=1");
+        xml.assertXpath("count(//Match)=1");
+        xml.assertXpath("count(//Match/Class[@name='MyClass'])=1");
+        xml.assertXpath("count(//Match/Method[@name='myMethod'])=1");
+        xml.assertXpath("count(//Match/Bug[@pattern='DM_EXIT'])=1");
+    }
+
+    /**
+     * To simplify the XML assertions.
+     */
+    static class ExcludesXml {
```

Comment:
@ilyakharlamov We already have a lib that does this xpath assertions, please use jcabi `XPathMatcher`

Reply:
@paulodamaso `jcabi-xml` is not currently a dependency of this project, I am not sure what's the policy about adding new dependencies, some architects allow new dependencies, some don't allow that, I am not sure.

Reply:
@ilyakharlamov I'm OK with any new dependency

Reply:
> ...please use jcabi `XPathMatcher`

@paulodamaso There is no such thing as jcabi XPathMatcher
there is a static `com.jcabi.matchers.XhtmlMatchers.hasXPath`, there is also a `com.jcabi.matchers.XPathMatcher` but it's package-private so it's not possible to use from `qulice`. So I don't understand, what do you want me to do?

Reply:
@ilyakharlamov Looks like you can use `XhtmlMatchers` instead. I know it's static and ugly, but I think that it's better the all this implementations. Feel free to fill a bug report on `jcabi-matchers` or `cactoos-matchers` to provide some elegant means of asserting an xparth value.